ecl/plugins/org.eclipse.rcptt.ecl.core/src/org/eclipse/rcptt/ecl/internal/commands/IfService.java

```diff
@@ -38,8 +41,14 @@ public IStatus service(Command command, IProcess context)
 		if (branch == null) {
 			return Status.OK_STATUS; // nothing to do
 		}
-		return session.execute(branch, context.getInput(), context.getOutput())
+		return session.execute(wrapBody(branch), context.getInput(), context.getOutput())
 				.waitFor();
 	}
 
+	private static Command wrapBody(Command body) {
```

Comment:
This could make a one more positive thing, if we additionally could wrap "if expression value" to some local variable of let.